src/util.ts

```diff
@@ -37,84 +37,95 @@ export function validatePlatformName(platform: string): boolean {
+	const targets = await fetchBuildTargets();
+	const target = targets.find((t) => t.version === requestedVersion);
+	if (!target) {
+		throw new Error(`Device OS version '${requestedVersion}' does not exist`);
 	}
-	if (!dvos[platform]) {
-		throw new Error(`Device OS version '${version}' does not support platform '${platform}'`);
+	if (!isSupportedPlatform(target, platform)) {
+		throw new Error(`Device OS version '${requestedVersion}' does not support platform '${platform}'`);
 	}
 	return true;
 }
 
-// Incomplete type definition for firmware manifest
-export interface FirmwareManifestV1 {
-	defaultVersions: { [key: string]: string };
-	binaryDataDeviceOS: { [key: string]: { [key: string]: string } };
+export interface BuildTargetV1 {
+	firmware_vendor: string;
+	platforms: number[];
+	prereleases: number[];
+	version: string;
 }
 
-let firmwareManifest: FirmwareManifestV1;
+export interface BuildTargetsResponseV1 {
+	targets: BuildTargetV1[];
+	platforms: { [key: string]: number }; // platform name -> platform id
+	defaultVersions: { [key: number]: string }; // platform id -> default version
+}
+
+let buildTargets: BuildTargetV1[];
 
-export async function fetchFirmwareManifest(): Promise<FirmwareManifestV1> {
-	if (firmwareManifest) {
-		return firmwareManifest;
+export async function fetchBuildTargets(): Promise<BuildTargetV1[]> {
+	if (buildTargets) {
+		return buildTargets;
 	}
 	const client = new httpm.HttpClient('particle-compile-action');
 	const res: httpm.HttpClientResponse = await client.get(
-		'https://binaries.particle.io/firmware-versions-manifest.json'
+		'https://api.particle.io/v1/build_targets'
 	);
 	if (res.message.statusCode !== 200) {
-		throw new Error(`Error fetching firmware manifest: ${res.message.statusCode}`);
+		throw new Error(`Error fetching build targets: ${res.message.statusCode}`);
 	}
-	firmwareManifest = JSON.parse(await res.readBody());
-	return firmwareManifest;
+	const body = JSON.parse(await res.readBody()) as BuildTargetsResponseV1;
+	buildTargets = body.targets;
+	return buildTargets;
 }
 
-export async function resolveVersion(platform: string, version: string): Promise<string> {
-	if (!version) {
+export async function resolveVersion(platform: string, requestedVersion: string): Promise<string> {
+	if (!requestedVersion) {
 		throw new Error(`Device OS version is required`);
 	}
 
-	const manifest = await fetchFirmwareManifest();
-	const latest = manifest.defaultVersions[platform];
-
-	delete manifest.binaryDataDeviceOS.binaryUrlGithub;
-	delete manifest.binaryDataDeviceOS.binaryUrlApi;
-	const versions = Object.keys(manifest.binaryDataDeviceOS).sort();
+	const targets = await fetchBuildTargets();
+	const versions = targets
+		.filter((t: BuildTargetV1) => isSupportedPlatform(t, platform))
+		.filter((t: BuildTargetV1) => prerelease(t.version) === null)
+		.map((t: BuildTargetV1) => t.version)
+		.sort();
+	const latest = versions[versions.length - 1];
```

Comment:
As I am looking at this one more time, I think this tries to get the latest version (not necessarily LTS).

Reply:
Right now, this tool doesn't consume the `default_versions` info. There are a couple of other version identifiers like `latest`, `latest-lts` and semver ranges it supports as user input. Docs: https://github.com/particle-iot/compile-action/blob/9b364b5ea9cb5f6ab006d7c9befc27db9a0574f2/README.md?plain=1#L23-L31

I think it makes sense to use the default_versions though: we could add a `default` allowed value (and it would make sense for it to be the default). Does adding `default` alongside `latest` and `latest-lts` seem reasonable to you? are the version names clear enough?

Reply:
I implemented the above - it is an improvement as certain platforms would error previously (p2, trackerm) because there is no LTS (but the default value was use the latest LTS)